aws/internal/service/xray/waiter/status.go

```diff
@@ -0,0 +1,17 @@
+package waiter
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/xray"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+// EncryptionConfigStatus fetches the Encryption Config and its Status
+func EncryptionConfigStatus(conn *xray.XRay) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+
+		output, _ := conn.GetEncryptionConfig(&xray.GetEncryptionConfigInput{})
+
+		return output, aws.StringValue(output.EncryptionConfig.Status), nil
```

Comment:
Nit: To prevent potential panics, this should `nil` check against `output` and `output.EncryptionConfig` before dereferencing (using new constant `EncryptionConfigStatusUnknown = "Unknown"`):

```suggestion
		if output == nil || output.EncryptionConfig == nil {
			return output, EncryptionConfigStatusUnknown, nil
		}
		
		return output, aws.StringValue(output.EncryptionConfig.Status), nil
```